pages/_app.js

```diff
@@ -118,6 +119,19 @@ export default function App({ Component, pageProps }) {
     });
   }
 
+  function handleUpdateInventoryToy(newToyId) {
+    setUserStats((prevStats) => {
+      const updatedInventory = { ...prevStats.inventory };
+      const toyIndex = updatedInventory.toy.findIndex(
+        (item) => item.id === newToyId
+      );
+      if (toyIndex !== -1) {
+        updatedInventory.toy[toyIndex].purchased = true;
+      }
```

Comment:
Consider to ensure a  proper state updates to avoid direct mutation. 

```suggestion
      if (toyIndex !== -1) {
      
      // clone the toy array
      const updatedToys = [...updatedInventory.toy];
      updatedToys[toyIndex] = { ...updatedToys[toyIndex], purchased: true };

      // set the updated toys array back to the updated inventory
      updatedInventory.toy = updatedToys;
    }
```